database/repository/candle/candle.go

```diff
@@ -0,0 +1,196 @@
+	"github.com/thrasher-corp/gocryptotrader/database/repository/exchange"
+	"github.com/thrasher-corp/gocryptotrader/log"
+	"github.com/thrasher-corp/sqlboiler/boil"
+	"github.com/thrasher-corp/sqlboiler/queries/qm"
+	"github.com/volatiletech/null"
+)
+
+// Candle generic candle holder for modelPSQL & modelSQLite
+type Candle struct {
+	ID         string
+	ExchangeID string
+	Base       string
+	Quote      string
+	Interval   string
+	Tick       []Tick
+}
+
+// Tick holds each interval
+type Tick struct {
+	Timestamp time.Time
+	Open      float64
+	High      float64
+	Low       float64
+	Close     float64
+	Volume    float64
+}
+
+// Series returns candle data
+func Series(exchangeName, base, quote, interval string, start, end time.Time) (out Candle, err error) {
+	if exchangeName == "" || base == "" || quote == "" || interval == "" {
+		err = errors.New("exchange, base , quote, interval, start & end cannot be empty")
+		return
+	}
+
+	exchangeUUID, err := exchange.UUIDByName(exchangeName)
+	if err != nil {
+		return
+	}
+
+	uuidQM := qm.Where("exchange_id = ?", exchangeUUID.String())
+	baseQM := qm.Where("base = ?", base)
+	quoteQM := qm.Where("quote = ?", quote)
+	intervalQM := qm.Where("interval = ?", interval)
+	dateQM := qm.Where("timestamp between ? and ?", start, end)
+
+	if repository.GetSQLDialect() == database.DBSQLite3 {
+		retCandle, errC := modelSQLite.Candles(uuidQM, baseQM, quoteQM, intervalQM, dateQM).All(context.Background(), database.DB.SQL)
+		if errC != nil {
+			return out, errC
+		}
+		for x := range retCandle {
+			t, errT := time.Parse(time.RFC3339, retCandle[x].Timestamp)
+			if errT != nil {
+				return out, errT
+			}
+			out.Tick = append(out.Tick, Tick{
+				Timestamp: t,
+				Open:      retCandle[x].Open,
+				High:      retCandle[x].High,
+				Low:       retCandle[x].Low,
+				Close:     retCandle[x].Close,
+				Volume:    retCandle[x].Volume,
+			})
+		}
+	} else {
+		retCandle, errC := modelPSQL.Candles(uuidQM, baseQM, quoteQM, intervalQM, dateQM).All(context.Background(), database.DB.SQL)
+		if errC != nil {
+			return out, errC
+		}
+
+		for x := range retCandle {
+			out.Tick = append(out.Tick, Tick{
+				Timestamp: retCandle[x].Timestamp,
+				Open:      retCandle[x].Open,
+				High:      retCandle[x].High,
+				Low:       retCandle[x].Low,
+				Close:     retCandle[x].Close,
+				Volume:    retCandle[x].Volume,
+			})
+		}
+	}
+
+	out.ExchangeID = exchangeName
+	out.Interval = interval
+	out.Base = base
+	out.Quote = quote
+
+	return out, err
+}
+
+// Insert series of candles
+func Insert(in *Candle) error {
+	if database.DB.SQL == nil {
+		return database.ErrDatabaseSupportDisabled
+	}
+
+	ctx := boil.SkipTimestamps(context.Background())
+	tx, err := database.DB.SQL.BeginTx(ctx, nil)
+	if err != nil {
+		log.Errorf(log.DatabaseMgr, "Insert transaction being failed: %v", err)
+		return err
+	}
+
+	if repository.GetSQLDialect() == database.DBSQLite3 {
+		err = insertSQLite(ctx, tx, in)
```

Comment:
out of scope: consider utilizing an interface so we don't need to determine dialect.